src/stream/src/executor/backfill/cdc/state.rs

```diff
@@ -12,71 +12,171 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use anyhow::anyhow;
 use maplit::hashmap;
+use risingwave_common::row;
 use risingwave_common::row::Row;
-use risingwave_common::types::{JsonbVal, ScalarRefImpl};
+use risingwave_common::types::{JsonbVal, ScalarImpl, ScalarRefImpl};
 use risingwave_common::util::epoch::EpochPair;
 use risingwave_connector::source::external::{CdcOffset, DebeziumOffset, DebeziumSourceOffset};
 use risingwave_connector::source::{SplitId, SplitImpl, SplitMetaData};
 use risingwave_storage::StateStore;
 use serde_json::Value;
 
+use crate::common::table::state_table::StateTable;
 use crate::executor::{SourceStateTableHandler, StreamExecutorResult};
 
+/// Depending on how the table is created, we have two scenarios for CDC Backfill:
+/// 1. `CREATE TABLE xx WITH ("connector"= 'mysql-cdc', "database.name"='mydb', "table.name"='t1')`
+/// In this case, the cdc backfill executor will wraps the source executor, and maintain its state
+/// (a finish flag) in the source state table.
+///
+///
+/// 2. `CREATE TABLE xx FROM source TABLE 'mydb.t1'`
+/// In this case, we can have multiple Table jobs sharing a single cdc Source job.
+/// The cdc backfill executor will be an instance of the `StreamScan` operator and has its own state table
+/// schema: `table_id | backfill_finished | row_count | cdc_offset`
 pub enum CdcBackfillStateImpl<S: StateStore> {
-    Undefined,
-    SingleTable(SingleTableState<S>),
+    SingleTable(SingleBackfillState<S>),
+    MultiTable(MultiBackfillState<S>),
+}
```

Comment:
I also want to deprecate it. But we need to change the original `Table` job plan from `source -> mview` to `source -> stream_scan -> mview`, so that it can has its own state, technically it is feasible.

Reply:
+1. Currently we have 3 ways to use mysql-cdc:
- `create table xxx with (....)`
- `set cdc_backfill='true'` + `create table xxx with (...)`
- `set cdc_backfill=true` + `create source with (...)` + `create table from ...`

It is confused and not easy to explain to user which one to choose. Ideally we should only have 2 ways:
1. `create table xxx with (....)`
2. `create source with (...)` + `create table from ...`

1 and 2 can share the same implementation and 1 can be a syntax suger for 2 if only user only needs to cdc one table from the db.

Reply:
The question is whether we need to ensure compatibility to make existing jobs created via set `cdc_backfill='true'` + `create table xxx with (...)` to be compatible. IMO, we can keep it simple and avoid compatibility codes here since cdc_backfill is announced as experimental feature only.